src/components/PopupWithForm.js

```diff
@@ -1,20 +1,41 @@
-import Popup from './Popup.js';
+import Popup from "./Popup.js";
 
-class PopupWithForm extends Popup {
-    constructor(popupSelector, handleFormSubmit) {
-        super({ popupSelector});
-        this.popupForm = this._popupElement.querySelector('.popup__form');
-        this._handleFormSubmit = handleFormSubmit;
+export default class PopupWithForm extends Popup {
+  constructor(popupSelector, submitCallback) {
+    super(popupSelector);
+    this._submitCallback = submitCallback;
+    this._form = this._popup.querySelector("form");
+    this._inputs = this._form.querySelectorAll("input");
+    this._submitButton = this._form.querySelector(".modal__button");
+    this._defaultButtonText = this._submitButton.textContent;
+  }
 
+  _getInputValues() {
+    const values = {};
+    this._inputs.forEach((input) => {
+      values[input.name] = input.value;
+    });
+    return values;
+  }
+
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._form
  }


```